Source/SwiftLintFramework/Rules/MultilineParametersRule.swift

```diff
@@ -40,6 +40,7 @@ public struct MultilineParametersRule: ASTRule, OptInRule, ConfigurationProvider
         for structure in dictionary.substructure {
             guard
                 let structureOffset = structure.offset,
+                structure.typeName != nil,
```

Comment:
I don't think this is enough to fix the issue, as you can explicit use a type for the closure parameter:

```swift
func foo(param1: Int,
         param2: Bool,
         param3: @escaping (Int) -> Void = { (x: Int) in }) { }
```

Reply:
Yep. You're right. :'-( Do you see a not-so-dirty way to filter out these parameters within parameters?

Reply:
Can’t think about anything right now, but I vaguely remind dealing with this before. Maybe one or the rules has this implemented?

Reply:
I'll take a look. Thanks :-)

Reply:
I can check if the offset of the parameter isn't in the range of other parameters.

Reply:
Done. I've pushed another commit that checks the range of the parameter. If the parameter is in the bounds of another parameter, it rejects it. Something like this:  a 1, a 2, and a 3 would be
filtered out since they're inside parameter 3.

```
dictionary.substructure varParameters
----------------------------------------------------------------|
|        |---------| |---------| |----------------------------| |
|        | param 1 | | param 2 | | param 3: |a 1| |a 2| |a 3| | |
|        |---------| |---------| |----------------------------| |
----------------------------------------------------------------|
```